ships/0035-beta-api-changes.md

```diff
@@ -0,0 +1,158 @@
+
+In the spirit of the Kubernetes Deprecation [policies](https://kubernetes.io/docs/reference/using-api/deprecation-policy/), the beta API will come with the following support guarantees:
+
+1. _No breaking changes, such as removal of fields, will be introduced without introducing a new beta API version (ex - `v1beta2`)._
+2. _Beta API versions may be deprecated, however they will continue to be served and supported for a period of 9 months after the deprecation is announced._
+3. _The beta API will never be deprecated in favor of an `alpha` API (ex - `v1alpha2`)._
+4. _Minimum API lifetime is determined by the API stability level_
+
+Related to 4), in this document we propose that for Beta API versions, we provide coverage after deprecation for **9** months. We do not encourage to be based on releases quantity, per the faster cadence we have compared to Kubernetes.
+
+### API Changes
+
+1. In all resources, introduce a convention for objects with an array, where each element contains the `name` key. An existing example is `.spec.volumes`.
+2. In the `Build` resource, replace `.spec.source.credentials` and `.spec.output.credentials` with a single field(_key_). This will become `.spec.source.cloneSecret` and `.spec.output.pushSecret` and will reference a secret name in the current namespace.
+3. In the `BuildStrategy` resource(_namespace and cluster scope_), replace `.spec.buildSteps` in favor of `steps`. The same applies to the `BuildStep` Go type.
+4. In the `BuildStrategy` resource(_namespace and cluster scope_), replace the `.spec.buildSteps[]`items with customize fields, instead of the whole `corev1.Container` Go type.
+5. In the `Build` resource, replace the `build.build.dev/build-run-deletion` annotation in favor of `.spec.retention.atBuildDeletion`.
+6. In the `BuildRun` resource, rename `.status.latestTaskRunRef` to `.status.taskRunName`.
+7. In the `BuildRun` resource, consolidate `.spec.buildRef.name` and `.spec.buildSpec` into `.spec.build.name` and `.spec.build.spec`.
+8. Use `type discriminators` if an API object has some form of implied polymorphism. For example, source types or volume sources.
+
+### Deprecation
+
+1. In the `Build` resource, deprecate `.spec.sources`. We only allow `.spec.source`.
+2. In the `Build` resource, deprecate `.spec.dockerfile`.
+3. In the `Build` resource, deprecate `.spec.builder`.
+4. In the `Build` resource, deprecate `.spec.volumes[].description`.
+5. In the `Build`, `BuildStrategy` and `ClusterBuildStrategy` resource, remove the `status` subresource.
+6. In the `BuildRun` resource, remove the support for auto-generating a service account.
+
+### Supported Use Cases
+
+#### Story 1
+
+As a user, I can create a Shipwright _v1beta1_ CRD. Shipwright should be able to convert this into a _v1alpha1_ CRD and stored that version in ETCD.
+
+#### Story 2
+
+As an external Kubernetes controller, I should be able to operate on Shipwright _v1beta1_ or _v1alpha1_ CRD's, independently of the API stored version.
```

Comment:
I think you here also mean custom resources and not their definition?

Reply:
correct. Fixed.